webservice/export-catalogs.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+var  ensureDb = require('dbjs/valid-dbjs')
+  , memoize = require('memoizee');
+
+module.exports = memoize(function (db) {
+	// ensure db
+	ensureDb(db);
+
+	var catalogs = [];
+	// all enums
+	var enums = db.StringLine.extensions.toArray().filter(function (extension) {
+		return extension.meta && extension.members;
+	}).map(function (extension) {
+		var catalogItem = { name: extension.__id__ };
+		catalogItem.options = extension.members.toArray().map(function (option) {
+			return { name: option, label: extension.meta[option].label };
+		});
+		return catalogItem;
+	});
+	catalogs.push({ enums: enums });
+
+	// services + roles
+	var services = [];
+	var roles = [];
+	var bpRoles = {};
```

Comment:
I really thinks we should not encourage programmers to identify processingSteps with roles. At some point we might be asked to expose roles, and it is something different than steps. Let's stick to processingSteps.